web/src/engine/websites/MeianPlus.ts

```diff
@@ -0,0 +1,183 @@
+import { Tags } from '../Tags';
+import icon from './MeianPlus.webp';
+import { Chapter, DecoratableMangaScraper, type MangaPlugin, Manga, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Fetch, FetchWindowScript } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as R } from '../../i18n/ILocale';
+
+type APISeries = {
+    licences: APISerieDetails[];
+}
+
+type APISerie = {
+    licence: APISerieDetails
+}
+
+type APISerieDetails = {
+    id_licence: number,
+    titre_licence: string,
+    articles: {
+        ref: number,
+        titre: string,
+        ebook_statut: boolean
+    }[]
+}
+
+type APIEbook = {
+    success: boolean,
+    message: string,
+    ebook: {
+        max_page: number
+    }
+}
+
+type APIImages = {
+    images: {
+        ref: number,
+        key: string,
+        w: number,
+        h: number,
+        param: string
+    }[]
+}
+
+type BlockInfo = {
+    w: number,
+    h: number
+}
+
+/**
+ * A basic oAuth token manager with MeianPlus specific business logic
+ */
+class TokenProvider {
+
+    #token: string = null;
+
+    constructor(private readonly clientURI: URL) { }
+
+    /**
+     * Extract the token directly from the website (e.g., after login/logout through manual website interaction)
+     */
+    public async UpdateToken() {
+        try {
+            this.#token = await FetchWindowScript<string>(new Request(this.clientURI), `(async () => (decodeURIComponent((await cookieStore.get('token_meian_plus')).value).replaceAll('"', '')))();`) ?? null;
```

Comment:
An async function always return a `Promise` so it can never be `null`, therefore `?? null` has no effect

Reply:
In this special case I would prefer `JSON.parse` to get the string from the enclosed double quotes `"` instead of `replaceAll` so the intention is more clear (`replaceAll` could be confused with replacing more than the enclosing double quotes)

Reply:
To avoid script errors (handle `null`) this could be re-arranged to
```javascript
this.#token = JSON.parse(decodeURIComponent(await FetchWindowScript<string>(new Request(this.clientURI), `(async () => (await cookieStore.get('token_meian_plus')).value)();`) ?? null));
```